src/common/bustub_instance.cpp

```diff
@@ -91,11 +92,17 @@ auto BustubInstance::ExecuteSql(const std::string &sql) -> std::vector<std::stri
     std::cerr << "=== PLANNER ===" << std::endl;
     std::cerr << planner.plan_->ToString() << std::endl;
 
+    // Optimize the query.
+    bustub::Optimizer optimizer(*catalog_);
+    auto optimized_plan = optimizer.Optimize(&*planner.plan_);
```

Comment:
```suggestion
    auto optimized_plan = optimizer.Optimize(planner.plan_.get());
```

If this works, I think it would be clearer than `&*`